contributing/documentation/overview.rst

```diff
@@ -35,48 +59,79 @@ Let's imagine that you want to improve the installation chapter of the Symfony
 
 **Step 1.** Go to the official Symfony documentation repository located at
-`github.com/symfony/symfony-docs`_ and `fork the repository`_ to your personal
-account. This is only needed the first time you contribute to Symfony.
+`github.com/symfony/symfony-docs`_ and click on the **Fork** button to `fork the
+repository`_ to your personal account. This is only needed the first time you
+contribute to Symfony.
 
-**Step 2.** **Clone** the forked repository to your local machine (this
-example uses the ``projects/symfony-docs/`` directory to store the documentation;
-change this value accordingly):
+**Step 2.** **Clone** the forked repository to your local machine (this example
+uses the ``projects/symfony-docs/`` directory to store the documentation; change
+this value accordingly):
 
 .. code-block:: bash
 
     $ cd projects/
     $ git clone git://github.com/<YOUR GITHUB USERNAME>/symfony-docs.git
 
-**Step 3.** Switch to the **oldest maintained branch** before making any change.
-Nowadays this is the ``2.3`` branch:
+**Step 3.** Add the original Symfony docs repository as a "Git remote" executing
+this command:
 
 .. code-block:: bash
 
     $ cd symfony-docs/
-    $ git checkout 2.3
+    $ git remote add upstream https://github.com/symfony/symfony-docs.git
+
+If things went right, you'll see the following when listing the "remotes" of
+your project:
 
-If you are instead documenting a new feature, switch to the first Symfony
-version which included it: ``2.5``, ``2.6``, etc.
+.. code-block:: bash
+
+    $ git remote -v
+    origin  git@github.com:<YOUR GITHUB USERNAME>/symfony-docs.git (fetch)
+    origin  git@github.com:<YOUR GITHUB USERNAME>/symfony-docs.git (push)
+    upstream  https://github.com/symfony/symfony-docs.git (fetch)
+    upstream  https://github.com/symfony/symfony-docs.git (push)
 
-**Step 4.** Create a dedicated **new branch** for your changes. This greatly
-simplifies the work of reviewing and merging your changes. Use a short and
-memorable name for the new branch:
+The purpose of this step is to allow you work simultaneously on the official
+Symfony repository and on your own fork. You'll see this in action in a moment.
+
+**Step 4.** Create a dedicated **new branch** for your changes. Use a short and
+memorable name for the new branch (if you are fixing a reported issue, use
+``fix_XXX`` as the branch name, where ``XXX`` is the number of the issue):
 
 .. code-block:: bash
 
-    $ git checkout -b improve_install_chapter
+    $ git checkout -b improve_install_chapter upstream/2.3
+
+In this example, the name of the branch is ``improve_install_chapter`` and the
+``upstream/2.3`` value tells Git to create this branch based on the ``2.3``
+branch of the ``upstream`` remote, which is the original Symfony Docs repository.
+
+Fixes should always be based on the **oldest maintained branch** which contains
+the error. Nowadays this is the ``2.3`` branch. If you are instead documenting a
```

Comment:
Honestly, we should store values like 2.3 in a configuration file somewhere and have a build step fill this
in dynamically (even if that process is a simple PHP script we write). That's outside of the scope, but
this is going to bite us very soon :)